coverage/python.py

```diff
@@ -63,7 +63,10 @@ def get_python_source(filename):
 
     # Replace \f because of http://bugs.python.org/issue19035
     source = source.replace(b'\f', b' ')
-    source = source.decode(source_encoding(source), "replace")
+    try:
+        source = source.decode(source_encoding(source), "replace")
+    except Exception as e:
```

Comment:
I would much rather catch a more specific error, in a more specific place.  The exception is happening inside source_encoding, so why not put the except-clause in that function?

Reply:
This felt like a good place for catching generic decoding errors and not just the one in the linked issue. Of course, I could add a specific exception handling in `source_encoding` if that's preferable.

Reply:
The reason this is a bad place for the except is that `s.decode(..., "replace")` can't raise an exception. In that line of code, it's only the `source_encoding()` call that can fail.

Reply:
Oh but this could be fixed by just moving `source_encoding` on its own line and wrapping it with `try...except` 😜 . Anyway, I'll have a closer look as soon as I can.

Reply:
Right, I've had another look and the idea of adding checks to `_source_encoding_py2` is not very appealing. Not to mention that they would have to be duplicated in `_source_encoding_py3`.

Reply:
Well, _source_encoding_py2 is gone on master, we've dropped Python 2 support.

Reply:
Ah I've just realised that support for Python 3.5 is also being dropped, which is not ideal for our usage. This prompted me to test on Python 3.7 with the master branch of `coveragepy` and the problem does not present itself 🙁 .